src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/IndentBlockFormattingRule.cs

```diff
@@ -21,28 +17,29 @@ internal sealed class IndentBlockFormattingRule : BaseFormattingRule
     {
         internal const string Name = "CSharp IndentBlock Formatting Rule";
 
-        private readonly CachedOptions _options;
+        private readonly CSharpSyntaxFormattingOptions _options;
 
         public IndentBlockFormattingRule()
-            : this(new CachedOptions(null))
+            : this(CSharpSyntaxFormattingOptions.Default)
         {
         }
 
-        private IndentBlockFormattingRule(CachedOptions options)
+        private IndentBlockFormattingRule(CSharpSyntaxFormattingOptions options)
         {
             _options = options;
         }
 
         public override AbstractFormattingRule WithOptions(SyntaxFormattingOptions options)
         {
-            var cachedOptions = new CachedOptions(options.Options);
+            var newOptions = options as CSharpSyntaxFormattingOptions ?? CSharpSyntaxFormattingOptions.Default;
```

Comment:
yick.  should we throw if we ere not given a CSharpSyntaxFormattingOptions ?

Reply:
I'm not sure how this may be used right now. I was being conservative here and preserve the behavior.